### PR TITLE
(bar&progress) remove context dependency, use a channel instead.

### PR DESCRIPTION
**Pull request by alevinval**
- This makes the library compatible with older versions of Go.
- Update `.travis.yml` to add more versions of go (2 yrs old release)

### CODE REVIEW

File: progress_go1.7.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// New Progress instance, it orchestrates the rendering of progress bars.
+// It supports cancellation via Context.
+func NewWithCtx(ctx context.Context) *Progress {
+	p := New()
+	go func() {
+		<-ctx.Done()
+		p.Stop()
+	}()
+	return p
+}
```
Comment (alevinval) [inline progress_go1.7.go]: I am not sure how useful is it to provide this functionality. If someone requires it they could easily implement it like that. Any thoughts?

My motivation for isolating context usage down to here is to make the library compatible with older versions of Go.

Having to do New(nil) also feels a bit weird, hence the separate constructor.

**Comment by alevinval** [comment]
Not sure why travis build is not triggering.

File: progress_go1.7.go (hunk 1 above)
Comment (vbauerster) [inline progress_go1.7.go]: Thank you for your work. But using context for cancelation is de-facto standard. As far as I know, context package is available in Go 1.6, but under `golang.org/x/net/context` path. I don't like the idea of removing context package. As for `New(nil)`, I agree it is a bit weird, but I did this on purpose, so one thinks about cancelation beforehand. By the way I plan to make some significant changes and bump to v2, so maybe I'll make just `New()`.

**Comment by vbauerster** [comment]
`New()`, `optional context` implemented in v2

**Pull request closed without merging**